File: csharp/StrangeCharactersRefactoring/TestCharacters/CharacterParserTest.cs
using Characters;
using NUnit.Framework;

namespace TestCharacters;

public class CharacterParserTest
{
    [TestCase]
    public void FindCharacterByPath()
    {
        CharacterParser.Certainly_InitializeFromFile_AndStuff(null);
        var character = CharacterParser.EvaluatePath("/Jim/Eleven");
        Assert.AreEqual("Eleven", character?.FirstName);
    }
    
    [TestCase]
    public void FindCharacterByEmptyPath()
    {
        CharacterParser.Certainly_InitializeFromFile_AndStuff(null);
        var character = CharacterParser.EvaluatePath("");
        Assert.Null(character);
    }

    [TestCase]
    public void FindCharacterByPathWithFamilyName()
    {
        CharacterParser.Certainly_InitializeFromFile_AndStuff(null);
        var character = CharacterParser.EvaluatePath("/Wheeler:Karen/Wheeler:Nancy");
        Assert.AreEqual("Nancy", character?.FirstName);
    }
    
    [TestCase]
    public void FindNemesisByPath()
    {
        CharacterParser.Certainly_InitializeFromFile_AndStuff(null);
        var character = CharacterParser.EvaluatePath("/Joyce/Will{Nemesis}");
        Assert.AreEqual("Mindflayer", character?.FirstName);
    }
    
    [TestCase]
    public void FindNemesisByPathAndFamilyName()
    {
        CharacterParser.Certainly_InitializeFromFile_AndStuff(null);
        var character = CharacterParser.EvaluatePath("/Wheeler:Karen/Wheeler:Nancy{Nemesis}");
        Assert.AreEqual(null, character);
    }
    
    [TestCase]
    public void FindNothingByPathAndFamilyName()
    {
        CharacterParser.Certainly_InitializeFromFile_AndStuff(null);
        var character = CharacterParser.EvaluatePath("/Wheeler:Karen/Wheeler:George");
        Assert.AreEqual(null, character);
    }
}

File: java/src/test/java/codingdojo/CharacterParserTest.java
package codingdojo;

import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

public class CharacterParserTest
{
    @Test
    public void findCharacterByPath() throws IOException {
        CharacterParser.InitializeFromFile(null);
        var character = CharacterParser.evaluatePath("/Jim/Eleven");
        assertEquals("Eleven", character.firstName);
    }
    
    @Test
    public void findCharacterByEmptyPath() throws IOException {
        CharacterParser.InitializeFromFile(null);
        var character = CharacterParser.evaluatePath("");
        assertNull(character);
    }

    @Test
    public void FindCharacterByPathWithFamilyName() throws IOException {
        CharacterParser.InitializeFromFile(null);
        var character = CharacterParser.evaluatePath("/Wheeler:Karen/Wheeler:Nancy");
        assertEquals("Nancy", character.firstName);
    }
    
    @Test
    public void FindNemesisByPath() throws IOException {
        CharacterParser.InitializeFromFile(null);
        var character = CharacterParser.evaluatePath("/Joyce/Will{Nemesis}");
        assertEquals("Mindflayer", character.firstName);
    }
    
    @Test
    public void FindNemesisByPathAndFamilyName() throws IOException {
        CharacterParser.InitializeFromFile(null);
        var character = CharacterParser.evaluatePath("/Wheeler:Karen/Wheeler:Nancy{Nemesis}");
        assertEquals(null, character);
    }
    
    @Test
    public void FindNothingByPathAndFamilyName() throws IOException {
        CharacterParser.InitializeFromFile(null);
        var character = CharacterParser.evaluatePath("/Wheeler:Karen/Wheeler:George");
        assertEquals(null, character);
    }
}
